src/browser/provider/built-in/dedicated/firefox/create-temp-profile.ts

```diff
@@ -71,7 +72,52 @@ async function generatePreferences (profileDir: string, { marionettePort, config
         ]);
     }
 
+    mimeTypes.split(',').forEach(mimeType => {
+        const type = mimeType.split('/')[1];
+
+        prefs.push(`user_pref("browser.download.viewableInternally.typeWasRegistered.${type}", true);`);
+    });
+
     await writeFile(prefsFileName, prefs.join('\n'));
+
+    // NOTE: The definitions of actions are there https://searchfox.org/mozilla-release/source/netwerk/mime/nsIMIMEInfo.idl#115
+    const handlersFileName = path.join(profileDir, 'handlers.json');
+    const handlers         = {
```

Comment:
Aleksey told me that while we could generate such settings for all possible MIME types, the browser will display only a few of them in UI. Likely it means that others are ignored by the browsers. Also, some file extensions can be shared by different MIME types. So we decided to keep only well-known types here, instead of adding code that leads to unpredictable results.